src/boundaries/edgetracking.jl

```diff
@@ -0,0 +1,304 @@
+successful (it may fail if the `mapper` maps both states to the same basin of attraction,
+in which case a warning is raised).
+
+## Keyword arguments
+* `bisect_thresh = 1e-7`: The maximum (Euclidean) distance between the two returned states.
+
+## Description
+`pds` is a `ParallelDynamicalSystem` with two states. The `mapper` must be an
+`AttractorMapper` of subtype `AttractorsViaProximity` or `AttractorsViaRecurrences`.
+
+!!! info
+    If the straight line between `u1` and `u2` intersects the basin boundary multiple
+    times, the method will find one of these intersection points. If more than two attractors
+    exist, one of the two returned states may belong to a different basin than the initial
+    conditions `u1` and `u2`. A warning is raised if the bisection involves a third basin.
+"""
+function bisect_to_edge(pds::ParallelDynamicalSystem, mapper::AttractorMapper;
+    bisect_thresh=1e-6)
+    u1, u2 = current_states(pds)
+    idx1, idx2 = mapper(u1), mapper(u2)
+
+    if (idx1 == idx2)
+        if idx1 == -1
+            error("AttractorMapper returned label -1 (could not match the initial condition with any attractor).
+            Try changing the settings of AttractorsViaProximity or increasing bisect_thresh, diverge_thresh.")
+        else
+            @warn "Both initial conditions belong to the same basin of attraction.
+                Attractor label: $(idx1)
+                u1 = $(u1)
+                u2 = $(u2)"
+            return u1, u2, false
+        end
+    end
+    
+    distance = diffnorm(u1, u2)
+    while distance > bisect_thresh
+        u_new = (u1 + u2)/2
+        idx_new = mapper(u_new)
+        # slightly shift u_new if it lands too close to the boundary
+        retry_counter = 1
+        while (idx_new == -1) && retry_counter < 3 # ToDO: make kwarg
+            @warn "Shifting new point slightly because AttractorMapper returned -1"
+            u_new += bisect_thresh*(u1 - u2)
+            idx_new = mapper(u_new)
+            retry_counter += 1
+        end
+        # update u1 or u2
+        if idx_new == idx1
+            u1 = u_new
+        else
+            if idx_new != idx2
+                if idx_new == -1
+                    error("AttractorMapper returned label -1 (could not match the initial condition with any attractor.)
+                    Try changing the settings of AttractorsViaProximity or increasing bisect_thresh, diverge_thresh.")
+                else
+                    @warn "New bisection point belongs to a third basin of attraction."
+                end
+            end
+            u2 = u_new
+        end
+        distance = diffnorm(u1, u2)
+    end
+    return u1, u2, true
+end;
+
+function diffnorm(u1, u2)
+    d = zero(eltype(u1))
+    @inbounds for i in eachindex(u1)
+        d += (u1[i] - u2[i])^2
+    end
+    sqrt(d)
+end;
+
+diffnorm(pds::ParallelDynamicalSystem) = diffnorm(current_state(pds, 1),
+    current_state(pds, 2))
```

Comment:
```suggestion
function diffnorm(pds::ParallelDynamicalSystem)
    diffnorm(current_state(pds, 1), current_state(pds, 2))
end
```
if it takes more than 1 line might as well use the long syntax